File: algorithms/python/consecutiveNumbersSum/consecutiveNumbersSum.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/consecutive-numbers-sum/
# Author: Miao Zhang
# Date:   2021-03-15

class Solution:
    def consecutiveNumbersSum(self, N: int) -> int:
        res = 1
        for i in range(2, int(sqrt(2 * N)) + 1):
            if (N - i * (i - 1) // 2) % i == 0:
                res += 1
        return res

File: algorithms/cpp/consecutiveNumbersSum/consecutiveNumbersSum.cpp
// Source: https://leetcode.com/problems/consecutive-numbers-sum/
// Author: Miao Zhang
// Date:   2021-03-15

class Solution {
public:
    int consecutiveNumbersSum(int N) {
        int res = 0, sums = 0;
        for (int i = 1; sums < N; i++) {
            sums += i;
            if ((N - sums) % i == 0) res++;
        }
        return res;
    }
};
